custom_components/eo_mini/api.py

```diff
@@ -97,6 +97,45 @@ async def async_post_enable(self, address) -> list[dict]:
             headers={"Content-Type": "application/x-www-form-urlencoded"},
         )
 
+    async def async_charge_mode_disable(
+        self, mode: str, charge_options: dict
+    ) -> list[dict]:
+        "Disable the specified charge mode"
+        data = charge_options
+        data[mode] = 0
+        return await self._async_api_wrapper(
+            "post",
+            f"{self.base_url}/api/user/updateChargeOpts",
+            data=data,
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+        )
+
+    async def async_charge_mode_enable(
+        self, mode: str, charge_options: dict
+    ) -> list[dict]:
+        "Enable the specified charge mode"
+        data = charge_options
+        data[mode] = 1
```

Comment:
Having this value be the only difference between the methods could cause issues in the future if a bug is fixed in one method but not the other - perhaps these methods should be consolidated into a single 'update charge mode options' method and you pass the key and value you want to update?

Reply:
yep, I've made that change so there is just one function in `api.py` to call now with the key and value to do these updates, though I still have the passing in the dict / circular dependency issue as above